## Home task 5

1. Когда используется критерий Стьюдента, а когда Z –критерий?  
Задачи 2,3 решать вручную
2. Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм. 
3.Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.  
Задачу 4 решать с помощью функции.  
4.Есть ли статистически значимые различия в росте дочерей? Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165 Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160

In [86]:
import numpy as np

In [87]:
import pandas as pd

In [88]:
from statistics import NormalDist as nd

In [89]:
import scipy.stats as stats

### Задание 1

Когда используется критерий Стьюдента, а когда Z –критерий?

Если можно использовать параметрический тест (генеральная совокупность распределена нормально):  
    если известна дисперсия ГС - используется Z-критерий;  
    если дисперсия ГС неизвестна - использцется t-критерий.

### Задание 2

Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

In [90]:
mu = 17

In [91]:
n = 100
x = 17.5
d = 4

In [92]:
a = 0.05

H0: x = mu  
H1: x > mu

In [93]:
z = (x - mu) / ((d/n)**0.5)

In [94]:
z

2.5

In [95]:
z_a = nd().inv_cdf(0.95)

In [96]:
z_a

1.6448536269514715

Расчетное значние критерия (2.5) > табличного значения (1,64) => принимается альтернативная гипотеза H1 (средний размер шариков не равен 17)

### Задание 3

Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест)

In [97]:
A = pd.DataFrame([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [98]:
mu = 200

In [99]:
a = 0.01
n = 10

H0: x = mu  
H1: x <> mu

In [100]:
x = A.mean()

In [101]:
x

0    198.5
dtype: float64

In [102]:
s = (((A - x) ** 2).sum() / (n-1)) ** 0.5

In [103]:
s

0    4.453463
dtype: float64

In [104]:
t = (x - mu) / (s/(n**0.5))

In [105]:
t

0   -1.065107
dtype: float64

In [106]:
t_a = -3.25

Расчетное значние критерия (-1.065) лежит правее табличного значения (-3.25) => гипотеза H0 не отвергается (средний вес пачки печенья составляет 200 г.)

### Задание 4

Есть ли статистически значимые различия в росте дочерей? Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165 Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160

In [107]:
A = pd.DataFrame([172, 177, 158, 170, 178,175, 164, 160, 169, 165])

In [108]:
B = pd.DataFrame([173, 175, 162, 174, 175, 168, 155, 170, 160])

H0: mu_A = mu_B  
H1: mu_A <> mu_B  

In [109]:
a = 0.05

In [110]:
stats.ttest_ind(A, B, equal_var=True)

Ttest_indResult(statistic=array([0.24400587]), pvalue=array([0.81014671]))

Полученное p-value > a (0.81>0.05)     => H0 не отвергается (статистически значимые различия в росте дочерей отсутствуют)